src/simsopt/geo/curve_planar_elliptical_cylindrical.py

```diff
@@ -0,0 +1,172 @@
+    return out
+
+
+def rotations(curve, a, b, alpha_r, alpha_phi, alpha_z, dr):
+    #rotates curves around r,phi,z coordinates
+
+    z_rot = jnp.asarray(
+        [[jnp.cos(alpha_z), -jnp.sin(alpha_z), 0],
+         [jnp.sin(alpha_z), jnp.cos(alpha_z), 0],
+         [0, 0, 1]])
+
+    y_rot = jnp.asarray(
+        [[jnp.cos(alpha_phi), 0, jnp.sin(alpha_phi)],
+         [0, 1, 0],
+         [-jnp.sin(alpha_phi), 0, jnp.cos(alpha_phi)]])
+
+    x_rot = jnp.asarray(
+        [[1, 0, 0],
+         [0, jnp.cos(alpha_r), -jnp.sin(alpha_r)],
+         [0, jnp.sin(alpha_r), jnp.cos(alpha_r)]])
+
+    out = curve
+
+    # I think this is doing the z rotation, so use b
+    out = out.at[:, 2].set(out[:, 2] + b)  # move bottom of coil up to z=0
+    out = out @ y_rot @ x_rot @ z_rot  # apply rotation in this frame
+    out = out.at[:, 0].set(out[:, 0] + dr)  # dr is the major radius shift
+    out = out.at[:, 2].set(out[:, 2] - b)  # move bottom of coils back down
+
+    return out
+
+
+def convert_to_cyl(a):
+    #convert to cylindrical - I think this still works for elliptical coils
+    out = jnp.zeros(a.shape)
+    out = out.at[:, 0].set(jnp.sqrt(a[:, 0]**2 + a[:, 1]**2))  # R = X^2 + Y^2
+    out = out.at[:, 1].set(jnp.arctan2(a[:, 1], a[:, 0]))  # phi = arctan(Y/X)
+    out = out.at[:, 2].set(a[:, 2])  # z = z
+    return out
+
+
+def cylindrical_shift(a, dphi, dz):
+    #shifting in phi, z
+    out = jnp.zeros(a.shape)
+    out = out.at[:, 0].set(a[:, 0])
+    out = out.at[:, 1].set(a[:, 1]+dphi)
+    out = out.at[:, 2].set(a[:, 2]+dz)
+    return out
+
+
+def cyl_to_cart(a):
+    #cylindrical to cartesian
+    out = jnp.zeros(a.shape)
+    out = out.at[:, 0].set(a[:, 0] * jnp.cos(a[:, 1]))  # x = R*cos(phi)
+    out = out.at[:, 1].set(a[:, 0] * jnp.sin(a[:, 1]))  # y = R*sin(phi)
+    out = out.at[:, 2].set(a[:, 2])  # z = z
+    return out
+
+
+def gamma_pure(dofs, points, a, b):
+    xyz = dofs[0:3]
+    ypr = dofs[3:6]
+    g1 = xyz_cyl(a, b, points)  # generate elliptical parameterization
+    g2 = rotations(g1, a, b, ypr[0], ypr[1], ypr[2], xyz[0])  # rotate and translate in R
+    g3 = convert_to_cyl(g2)  # convert to R, phi, Z coordinates
+    g4 = cylindrical_shift(g3, xyz[1], xyz[2])  # shift in phi and Z
+    final_gamma = cyl_to_cart(g4)  # convert back to cartesian
+    return final_gamma
+
+
+class CurvePlanarEllipticalCylindrical(JaxCurve):
+    """
+    CurvePlanarEllipticalCylindrical is a translated and rotated Curve in r, phi, and z coordinates.
+    It is parametrized by an ellipse with semi-major axis a and semi-minor axis b, where a lies along 
+    the R vector and b along the Z vector. This curve is then translated and rotated in cylindrical coordinates
+    to a major radius R0, toroidal angle phi, and vertical position Z0. The curve is then rotated by angles
+    r_rotation, phi_rotation, and z_rotation in the r, phi, and z directions, respectively, about the minimum 
```

Comment:
what does it mean to rotate by these angles? Are you rotating about the local cylindrical basis axes?